cirq/protocols/json.py

```diff
@@ -132,12 +143,33 @@ def cirq_class_resolver_dictionary(self) -> Dict[str, Type]:
                 'sympy.Rational': sympy.Rational,
                 'complex': complex,
             }
+            self._crd.update(third_party)
         return self._crd
 
 
 RESOLVER_CACHE = _ResolverCache()
 
 
+def AddJson(Cls=None, *, key=""):
```

Comment:
Technically I think this should be in snake case, ie `add_json(cls=...)`. Or maybe "`with_json`" so that it reads nicer as a decorator.

It might be nice to have `key` default to the class name